File: adm-vitrine/src/app/modules/order/model/payload/order-create.payload.ts
import {Payload} from '@shared/model';
import {AddressDto, ProfileDto} from '@account/model';
import {OrderLineUpdatePayload, OrderStatus} from '@order/model';

export interface OrderCreatePayload extends Payload {
  reference: string;
  profile: Partial<ProfileDto>;
  status: OrderStatus;
  comment: string;
  lines: OrderLineUpdatePayload[];
  billingAddress: AddressDto;
  delivryAddress: AddressDto
}

File: api/src/main/java/io/studerd/api/module/order/entity/payload/OrderCreatePayload.java
package io.studerd.api.module.order.entity.payload;

import io.studerd.api.module.address.entity.Address;
import io.studerd.api.module.order.entity.OrderLine;
import io.studerd.api.module.profile.entity.Profile;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class OrderCreatePayload {
    private String reference;
    private Profile profile;
    private String status;
    private String comment;
    private List<OrderLine> lines;
    private Address billingAddress;
    private Address delivryAddress;
}
